docs/DesignDocs/SGX_EMM.md

```diff
@@ -0,0 +1,679 @@
+SGX Enclave Memory Manager
+=================================
+
+## Motivation ##
+
+An enclave's memory is backed by a special reserved region in RAM, called
+Enclave Page Cache (EPC). Enclave memory management tasks include
+allocating/reserving virtual address ranges, committing physical EPC pages,
+changing EPC page permissions or page types, and removing EPC pages.
+Those tasks require collaboration between the trusted runtime, the untrusted
+runtime, and the OS. The SGX enclave memory manager (EMM) serves as a central
+component in the enclave trusted runtime that abstracts the interaction with
+the untrusted runtime for all memory management flows and provides APIs for
+its clients to reserve virtual address ranges, commit EPC memory to the reserved
+address ranges, and modify attributes of the reserved/committed pages.
+
+For details of specific memory management related flows, please refer to
+[the SGX EDMM driver API spec](https://github.com/openenclave/openenclave/pull/3755/files).
```

Comment:
design docs shouldn't refer to pull requests, only other docs

Reply:
will resolve this after driver interface merged